addons/play-later/addToListButton.js

```diff
@@ -0,0 +1,52 @@
+import { addToList, getList, removeFromList } from './list.js';
+
+function projectId() {
+    return window.location.pathname.split('projects/')[1].split('/')[0];
+}
+
+async function addToStorage() {
+    addToList(projectId());
+}
+
+async function removeFromStorage() {
+    removeFromList(projectId());
+}
+
+export default async function ({ addon, console, msg }) {
+    async function createButton() {
+        await addon.tab.waitForElement('.project-buttons');
+
+        let existingButton = document.querySelector('.sa-play-later-button');
+        if (existingButton) {
+            existingButton.remove();
+        }
+
+        const isRemoveButton = getList().includes(projectId());
+        let button = document.createElement('button');
+        button.classList.add('button', 'action-button', 'sa-play-later-button');
+        if (isRemoveButton) {
+            button.textContent = msg('remove');
+        } else {
+            button.textContent = msg('add');
+        }
+        
+        button.addEventListener('click', () => {
+            if (isRemoveButton) {
+                removeFromStorage();
+            } else {
+                addToStorage();
+            }
+            
+            createButton();
+        });
+
+        addon.tab.appendToSharedSpace({space: 'afterCopyLinkButton', element: button});
+    }
+
+    await createButton();
+    addon.tab.addEventListener('urlChange', () => {
```

Comment:
Ah, that makes sense. Is this required though? I ask because this case seems different from the other ones, since the button content and function change each time it is pressed. I can make this change if it fits a standard better, but I don't think it will be less code.